backend/src/controller/product.ts

```diff
@@ -83,25 +83,27 @@ module.exports.deleteProduct = async (req: Request, res: Response) => {
     }
 };
 
-module.exports.editProduct = (req: Request, res: Response) => {
-    if (!req.body) {
+module.exports.editProduct = async (req: Request, res: Response) => {
+    try {
+        const foundProduct = await product.findById(req.body.productID);
+        req.body.productName
```

Comment:
suggestion: since keys in the object and `req.body` are the same, you can probably use the `for...in` loop like so:

```
for (const key in foundProduct) {
    foundProduct[key] = req.body[key] || foundProduct[key]
}
```

You'd have to check how it behaves with the keys that come with the mongoose schema, worst case, you'll have to check if the key is something that the user can change.